docs/leaderboard/develop/api-reference/index.md

```diff
@@ -0,0 +1,193 @@
+---
+sidebar_position: 1
+sidebar_label: API Reference
+title: Leaderboard API reference
+description: Learn how to interact programmatically with Momento Leaderboard API.
+---
+
+import { SdkExampleTabs } from "@site/src/components/SdkExampleTabs";
+// This import is necessary even though it looks like it's un-used; The inject-example-code-snippet
+// plugin will transform instances of SdkExampleTabs to SdkExampleTabsImpl
+import { SdkExampleTabsImpl } from "@site/src/components/SdkExampleTabsImpl";
+
+## Using the Momento Leaderboard
+
+Momento Leaderboard is a first-class service with purpose-built APIs that supports leaderboards with tens of millions of items and rapid ingestion/querying/updates.
+
+## Leaderboard Client
+
+To create and interact with leaderboards, you must first create a LeaderboardClient.
+
+<SdkExampleTabs snippetId={'API_InstantiateLeaderboardClient'} />
+
+Then you can create a Leaderboard by specifying a cache and leaderboard name.
+
+<SdkExampleTabs snippetId={'API_CreateLeaderboard'} />
+
+## Leaderboard Methods
+
+### Upsert elements
+
+Inserts elements if they do not already exist in the leaderboard. Updates elements if they do already exist in the leaderboard.
+
+| Name         | Type                | Description                                                    |
+|--------------|---------------------|----------------------------------------------------------------|
+| elements     | Map<number, number> | A Map or Record of elements (id,score pairs) to upsert.        |
```

Comment:
Shouldn't this be `Map<string, number>`?

Reply:
No, the [protos](https://github.com/momentohq/client-protos/blob/main/proto/leaderboard.proto#L61) are expecting a `number` for the unique identifier